controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -200,6 +212,32 @@ func markKubernetesVersionCondition(oacp *controlplanev1alpha2.OpenshiftAssisted
 	}
 }
 
+func IsUpgradeRequested(ctx context.Context, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) bool {
```

Comment:
Since we've made it public, I think we should create a package (or find a suitable one) for this and keep the controller package slimmer -same goes for GetWorkloadClusterVersion. WDYT?

Reply:
Good idea! Moved it out!